internal/services/storage/storage_share_resource.go

```diff
@@ -222,6 +236,10 @@ func resourceStorageShareRead(d *pluginsdk.ResourceData, meta interface{}) error
 	d.Set("url", id.ID())
 	d.Set("enabled_protocol", string(props.EnabledProtocol))
 
+	if accessTier := props.AccessTier; accessTier != nil {
+		d.Set("access_tier", string(*accessTier))
+	}
```

Comment:
this'll want a value always set:

```suggestion
	accessTier := ""
	if props.AccessTier != nil {
		accessTier = string(*props.AccessTier)
	}
	d.Set("access_tier", accessTier)
```